File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int L = 0, R = nums.length - 1, mid;

        while (L <= R) {
            mid = (L + R) / 2;
            
            if (target < nums[mid]) {
                R = mid - 1;
            } else if (target > nums[mid]) {
                L = mid + 1;
            } else {
                return mid;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/sort-colors/submission-0.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        helper(nums, 0, nums.size() - 1);
    }

private:
    void helper(vector<int>& nums, int s, int e) {
        if(e - s + 1 <= 1) return;

        int pivot = nums[e], left = s;

        for(int i = s; i < e; i++) {
            if (nums[i] < pivot) swap(nums[left++], nums[i]);
        }

        nums[e] = nums[left];
        nums[left] = pivot;

        helper(nums, s, left - 1); helper (nums, left + 1, e);
    }
};